smirnoff_plugins/handlers/bonded.py

```diff
@@ -0,0 +1,120 @@
+import abc
+
+from openff.toolkit import unit
+from openmm import CustomCompoundBondForce
+from openff.toolkit.typing.engines.smirnoff.parameters import (
+    ElectrostaticsHandler,
+    IncompatibleParameterError,
+    LibraryChargeHandler,
+    ParameterAttribute,
+    ParameterHandler,
+    ParameterType,
+    ToolkitAM1BCCHandler,
+    VirtualSiteHandler,
+    _allow_only,
+    vdWHandler,
+    ImproperTorsionHandler,
+    BondHandler
+)
+
+
+class _CustomBondedHandler(ParameterHandler, abc.ABC):
+    """Base class for custom bonded parameter handlers (bonds, angles, torsions)."""
+
+    cutoff = ParameterAttribute(default=9.0 * unit.angstroms, unit=unit.angstrom)
+    
+    periodicity = ParameterAttribute(
+        default=1, converter=int
+    )
+
+    periodic_method = ParameterAttribute(
+        default="cutoff", converter=_allow_only(["cutoff"])
+        )
+    
+    nonperiodic_method = ParameterAttribute(
+        default="no-cutoff", converter=_allow_only(["no-cutoff"])
+        )
+
+    switch_width = ParameterAttribute(default=1.0 * unit.angstroms, unit=unit.angstrom)
```

Comment:
the header needs to be changed from nonbonded to your custom handler's harmonic form